frontend/src/components/popup.tsx

```diff
@@ -0,0 +1,54 @@
+import clsx from 'clsx';
+import { cloneElement, ReactElement, ReactNode, useState } from 'react';
+import { usePopper } from 'react-popper';
+
+export function Popup({ children, button }: { children?: ReactNode; button: ReactElement }) {
+  const [referenceElement, setReferenceElement] = useState<HTMLDivElement | null>(null);
+  const [popperElement, setPopperElement] = useState<HTMLDivElement | null>(null);
+
+  const { styles, attributes, update } = usePopper(referenceElement, popperElement, {
+    placement: 'bottom',
+    modifiers: [
+      {
+        name: 'flip',
+        options: {
+          fallbackPlacements: ['top'],
+          flipVariations: true,
+        },
+      },
+    ],
+  });
+  const [show, setShow] = useState(false);
+
+  return (
+    <div ref={setReferenceElement}>
+      {cloneElement(button, {
+        onClick: () => {
+          setShow((old) => !old);
+          update && update();
+        },
+      })}
+
+      {show ? (
+        <div
+          className={clsx(
+            'p-4',
+            'bg-white',
+            'border-black',
+            'border-2',
+            'shadow-brutal',
+            'rounded-lg',
+            'm-2',
```

Comment:
Popper does not like margins on itself (I think this line is was causes it). The popper element reports in the console:

> CSS "margin" styles cannot be used to apply padding between the popper and its reference element or boundary. To replicate margin, use the `offset` modifier, as well as the `padding` option in the `preventOverflow` and `flip` modifiers. 

More Details: https://popper.js.org/docs/v2/migration-guide/#4-remove-all-css-margins

Reply:
ah good catch!